packages/ragbits-cli/src/ragbits/cli/app.py

```diff
@@ -0,0 +1,60 @@
+import json
+from dataclasses import dataclass
+from typing import Any
+
+import typer
+from pydantic import BaseModel
+from rich.console import Console
+from rich.table import Table
+
+
+@dataclass()
+class CliState:
+    """A dataclass describing CLI state"""
+
+    output_type: str = "text"
+
+
+class CLI(typer.Typer):
+    """A CLI class with output formatting"""
+
+    def __init__(self, *args: Any, **kwargs: Any):  # noqa: ANN401
+        super().__init__(*args, **kwargs)
+        self.state: CliState = CliState()
+        self.console: Console = Console()
+
+    def set_output_type(self, output_type: str) -> None:
+        """
+        Set the output type in the app state
+        Args:
+            output_type: str
+        Raises:
+            ValueError - if the output_type is not `json` or `text`
+        """
+        if output_type not in ["text", "json"]:
+            raise ValueError("Output type must be either 'text' or 'json'")
+        self.state.output_type = output_type
+
+    def print_output(self, data: list[BaseModel]) -> None:
```

Comment:
I feel it's likely that we will want this to become `print_output(self, data: list[BaseModel] | BaseModel)` and also to be able to take additional configuration regarding how things are displayed, but I think this can be left for the future.

Reply:
What I mean is that while `list[BaseModel]` should be displayed in a table maybe `BaseModel` could be displayed vertically by default instead - this would give us good support for both commands that list objects and commands that display details of a single object.

As to options - they could allow for more control of how things are laid out and which fields are included.

But as I said: this could become a separate ticket. We will know more when we start to add more commands.

Reply:
ok - will be left for the future. for now i've just allowed `BaseModel` to be used as input